src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -46,6 +49,20 @@ private static unsafe bool IsValidAddress(ReadOnlySpan<char> ipSpan)
             return false;
         }
 
+        private static unsafe bool IsSafeDnsString(ReadOnlySpan<char> name)
+        {
+            for (int i = 0; i < name.Length; i++)
```

Comment:
This could use the new `IndexOfAnyValues<T>`, but not sure if the allocation of static map is worth it for such short strings.

Reply:
Yes, this should use `IndexOfAnyValues`. The allocation there is very small, we'll only store a bitmap for the ASCII range.
```c#
// Alphanumeric or '-', '.', '_'
private static readonly IndexOfAnyValues<char> s_safeDnsChars =
    IndexOfAnyValues.Create("-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz");

private static bool IsSafeDnsString(ReadOnlySpan<char> name) =>
    name.IndexOfAnyExcept(s_safeDnsChars) >= 0;
```

Reply:
`IndexOfAnyValues` crossed my mind but the set of characters I don't want is large for Unicode. I was not aware of `IndexOfAnyExcept`

Reply:
updated. Just to be sure: this would only match that ASCII characters, not their Unicode equivalent (if any), right @MihaZupan ?

Reply:
That's right, the set of values you pass to `Create` is "ordinal".